src/shared_library.c

```diff
@@ -0,0 +1,93 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+#include "rcutils/shared_library.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+
+rcutils_shared_library_t *
+rcutils_get_shared_library(const char * library_path)
+{
+  rcutils_shared_library_t * lib = (rcutils_shared_library_t *)
+    malloc(sizeof(rcutils_shared_library_t));
```

Comment:
Should this support a custom allocator?

Same below.

Reply:
👍, if allocation cannot be avoided, then an `rcutils_allocator_t` should be one of the arguments.